AirCasting/CreateSessionViews/SelectDeviceView.swift

```diff
@@ -93,7 +93,19 @@ struct SelectDeviceView: View {
                     isTurnOnBluetoothLinkActive = true
                 }
             } else if selected == 2 {
-                isMicLinkActive = true
+                if microphoneManager.audioSession.recordPermission == .granted {
+                    isMicLinkActive = true
+                } else {
+                    microphoneManager.audioSession.requestRecordPermission { (isGranted) in
```

Comment:
I'm not sure how I can move `requestRecordPermission` function out of this view if I need to put the view logic inside the closure in this function. Otherwise I'm not sure how the app can wait for the user's choice before moving to another screen/going to settings.